src/routes/Detail.js

```diff
@@ -1,9 +1,51 @@
 import React from 'react';
+import RepositoryDetailModal from 'components/RepositoryDetailModal';
+import axios from 'axios';
 
 
 export default class Detail extends React.Component {
+  state = { isOpen: false };
+
+  static contextTypes = {
+    router: React.PropTypes.func.isRequired
+  }
+  open = () => {
+    this.setState({ repository: null, isOpen: true });
+  }
+  close = () => {
+    this.setState({ isOpen: false });
+    this.context.router.goBack();
+  }
```

Comment:
I wouldnt' use `goBack`, since you may have nothing in the history (i.e. start navigating from `search/detail/angular/angular?query=angular`).
I would use `transitionTo('search, { query: ... `